### PR TITLE
ANN: Reduce the error scope for E0130, E0561 and E0642 from value parameter to pattern

### PR DESCRIPTION
**Pull request by missingdays**

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt [hunk 1]
```diff
@@ -321,30 +322,68 @@ private fun isComplexPattern(pat: RsPat?): Boolean {
...
+        }
+
+        is RsAbstractableOwner.Trait -> {
+            if (isComplexPattern && fn.block == null) {
+                RsDiagnostic.PatternArgumentInFunctionWithoutBodyError(element).addToHolder(holder)
+            }
+        }
+
+        else -> {}
+    }
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt:366]: I think after extraction from `checkValueParameterInFunction` is can be simplified:

```suggestion
    val owner = fn.owner
    val isComplexPatternForbidden = owner is RsAbstractableOwner.Foreign
        || owner is RsAbstractableOwner.Trait && fn.block == null

    if (isComplexPatternForbidden && isComplexPattern(element)) {
        RsDiagnostic.PatternArgumentInForeignFunctionError(element).addToHolder(holder)
    }
```

Comment (missingdays) [reply src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt:366]: But if `fn.owner` is `RsAbstractableOwner.Trait`, `RsDiagnostic.PatternArgumentInFunctionWithoutBodyError` is emitted, not `RsDiagnostic.PatternArgumentInForeignFunctionError`

Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt:366]: Aah, damn! Then nevermind :)

**Review by vlad20012 — APPROVED** [review]
bors r+

**Pull request merged**